src/vaadin-context-menu.html

```diff
@@ -642,12 +642,25 @@
                 style.right = right - padding + 'px';
                 alignedToParent = true;
               }
+            } else if (x < parentContentRect.x) {
+              // Check if sub menu opens on the left side and the parent menu is not right aligned.
+              // If so, use actual width of the submenu content instead of the parent menu content.
+              x = x - (width - parentContentRect.width);
             }
           }
 
           if (!alignedToParent) {
-            if (x < wdthVport / 2 || x < xMax) {
+            if ((x < wdthVport / 2 || x < xMax) && !parent) {
               style.left = x + 'px';
+            } else if ((x < wdthVport / 2 || x < xMax) && parent &&
+              wdthVport - parent.$.overlay.getBoundingClientRect().width - parent.$.overlay.getBoundingClientRect().x
```

Comment:
There is already `const parentContentRect = parent.$.overlay.getBoundingClientRect();` (line 629).

Let's update lines 626 - 629 to look like this:

```js
const parent = overlay.parentOverlay;
let alignedToParent = false;
let parentContentRect;
if (parent) {
  parentContentRect = parent.$.overlay.getBoundingClientRect();
```

Then you can use `parentContentRect` below in all the places where you check it.

Reply:
done